api/src/gameState/gameState.controller.ts

```diff
@@ -38,9 +39,44 @@ export class GameStateController {
   }
 
   @Get('')
-  async getGames(): Promise<{ data: Game[] }> {
+  async getGames(
+    @Query('sort') sort: string,
+    @Query('range') range: string,
+  ): Promise<{ data: Game[] }> {
+    let params;
+
+    if (range) {
+      try {
+        const rangeNumber = JSON.parse(range) as number[];
+        if (rangeNumber) {
+          params = {
+            ...params,
+            take: rangeNumber[1] - rangeNumber[0],
+            skip: rangeNumber[0],
+          };
+        }
+      } catch (e) {
+        throw new HttpException('Incorrect range parameter', 400);
+      }
+    }
+
+    if (sort) {
+      try {
+        const sortParsed = JSON.parse(sort);
+        if (sortParsed) {
+          params = {
+            ...params,
+            orderBy: {
+              [sortParsed[0]]: sortParsed[1].toLowerCase(),
+            },
+          };
+        }
+      } catch (e) {
+        throw new HttpException('Incorrect sort parameter', 400);
```

Comment:
use the dedicated exception instead of the generic one. It is easier to read BadRequestException instead of HttpException....